experiments/timeline-benchmark/README.md

```diff
@@ -54,14 +54,48 @@ and actual question is:
 
 where `t` and `c` (test and control, or test and corrext answer) are the lines.
 
-benchmark.js expects a chat completion style API available at: `http://127.0.0.1:8080/completion`. If you run llamafile (I test v1.5), and mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf, you start the server:
+benchmark.js expects a chat completion style API available at: `http://127.0.0.1:8080/completion`. If you run llamafile (I test v1.5), and mixtral-8x7b-instruct-v0.1.Q5_K_M.gguf, you start the server:
 
-    # ./llava-v1.5-7b-q4.llamafile -m mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf
+    # ./llava-v1.5-7b-q4.llamafile -m mixtral-8x7b-instruct-v0.1.Q5_K_M.gguf
 
 and then run the benachmark:
 
     # node benchmark.js test.txt expected.txt
 
 produces three numbers: (1) total number of correct entries, (2) entries missed in `test.txt`, and (3) excess entries that LLM produced.
 
-Currently test.txt has three entries with a duplicate. and expected.txt has two. the order is not sorted. Some LLM may produce a date as "Undated" or "1960's", In any case, we want to check all possibilities so we do NxM comparison at this moment.
+Currently test.txt has three entries that are duplicate. and expected.txt has two. the order is not sorted. Some LLM may produce a date as "Undated" or "1960's", In any case, we want to check all possibilities so we do NxM comparison at this moment.
+
+At the end of NxM judgements, the program produces a line that looks like this:
+
+   `total = 20, hit = 29, missed = 0`
+
+This means that there are 20 entries in the "expected" file, and they are 29 entries in the "test" file that have the equivalent entry in "expected". Because there are duplicates that "may" be considered equivalent, the hit in this case can be more than `total`. The `missed` value shows how many of  `expected` entries are not hit at all by the "test" entries.
+
+## A test file with some amgibious entries.
+
+`ambigious.txt has some overlapping concepts in it. As above, run an LLM in the server mode:
+
+    # ./llava-v1.5-7b-q4.llamafile -m mixtral-8x7b-instruct-v0.1.Q5_K_M.gguf
+
+and then run:
+
+    # node benchmark.js ambigious.txt ambigious.txt
+
+so that the "test" and "expected" files are the same. An expectation is to get "hit" somewhere between 20-26, and hit should be an even number.
+
+## A less ambigious test
+
+`less-ambigious.txt` has fewer entries than `amgibious.txt`. If you run:
+
+    # node benchmark.js less-ambigious.txt less-ambigious.txt
+
+In this case, the expection is that only the diagonal pairs in the NxN matrix (where N=11)are "yes" as they are the same, and everything else is "no". So the expected output is:
+
+    `total = 11, hit = 11, missed = 0`
+
+For my trials, mixtral-8x7b-instruct-v0.1.Q5_K_M.gguf gives some wrong answers.
```

Comment:
Can you include the outputs that this version of mixtral emits?

Reply:
```
ohshima@Asshimar timeline-benchmark % node benchmark.js less-ambigious.txt less-ambigious.txt 
yes 1965: Butler Lampson's thesis. 1965: Butler Lampson's thesis.
no 1965: Butler Lampson's thesis. 1966: Simula introduced by Nygaard and Dahl.
yes 1965: Butler Lampson's thesis. 1967: Case Western Reserve University founded.
no 1965: Butler Lampson's thesis. 1969: ARPANET started running.
no 1965: Butler Lampson's thesis. 1970: Xerox PARC founded.
no 1965: Butler Lampson's thesis. 1972: Smalltalk created.
no 1965: Butler Lampson's thesis. 1979: C++ development started by Bjarne Stroustrup.
no 1965: Butler Lampson's thesis. 1981: MS-DOS introduced.
yes 1965: Butler Lampson's thesis. 1984: X Windows introduced.
no 1965: Butler Lampson's thesis. 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee.
no 1965: Butler Lampson's thesis. 1995: Java programming language and Java VM first released by Sun Microsystems.
no 1966: Simula introduced by Nygaard and Dahl. 1965: Butler Lampson's thesis.
yes 1966: Simula introduced by Nygaard and Dahl. 1966: Simula introduced by Nygaard and Dahl.
no 1966: Simula introduced by Nygaard and Dahl. 1967: Case Western Reserve University founded.
no 1966: Simula introduced by Nygaard and Dahl. 1969: ARPANET started running.
no 1966: Simula introduced by Nygaard and Dahl. 1970: Xerox PARC founded.
no 1966: Simula introduced by Nygaard and Dahl. 1972: Smalltalk created.
no 1966: Simula introduced by Nygaard and Dahl. 1979: C++ development started by Bjarne Stroustrup.
no 1966: Simula introduced by Nygaard and Dahl. 1981: MS-DOS introduced.
no 1966: Simula introduced by Nygaard and Dahl. 1984: X Windows introduced.
no 1966: Simula introduced by Nygaard and Dahl. 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee.
no 1966: Simula introduced by Nygaard and Dahl. 1995: Java programming language and Java VM first released by Sun Microsystems.
no 1967: Case Western Reserve University founded. 1965: Butler Lampson's thesis.
no 1967: Case Western Reserve University founded. 1966: Simula introduced by Nygaard and Dahl.
yes 1967: Case Western Reserve University founded. 1967: Case Western Reserve University founded.
no 1967: Case Western Reserve University founded. 1969: ARPANET started running.
no 1967: Case Western Reserve University founded. 1970: Xerox PARC founded.
no 1967: Case Western Reserve University founded. 1972: Smalltalk created.
no 1967: Case Western Reserve University founded. 1979: C++ development started by Bjarne Stroustrup.
no 1967: Case Western Reserve University founded. 1981: MS-DOS introduced.
no 1967: Case Western Reserve University founded. 1984: X Windows introduced.
no 1967: Case Western Reserve University founded. 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee.
yes 1967: Case Western Reserve University founded. 1995: Java programming language and Java VM first released by Sun Microsystems.
no 1969: ARPANET started running. 1965: Butler Lampson's thesis.
no 1969: ARPANET started running. 1966: Simula introduced by Nygaard and Dahl.
no 1969: ARPANET started running. 1967: Case Western Reserve University founded.
no 1969: ARPANET started running. 1969: ARPANET started running.
no 1969: ARPANET started running. 1970: Xerox PARC founded.
no 1969: ARPANET started running. 1972: Smalltalk created.
no 1969: ARPANET started running. 1979: C++ development started by Bjarne Stroustrup.
no 1969: ARPANET started running. 1981: MS-DOS introduced.
no 1969: ARPANET started running. 1984: X Windows introduced.
no 1969: ARPANET started running. 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee.
no 1969: ARPANET started running. 1995: Java programming language and Java VM first released by Sun Microsystems.
no 1970: Xerox PARC founded. 1965: Butler Lampson's thesis.
no 1970: Xerox PARC founded. 1966: Simula introduced by Nygaard and Dahl.
no 1970: Xerox PARC founded. 1967: Case Western Reserve University founded.
no 1970: Xerox PARC founded. 1969: ARPANET started running.
no 1970: Xerox PARC founded. 1970: Xerox PARC founded.
no 1970: Xerox PARC founded. 1972: Smalltalk created.
no 1970: Xerox PARC founded. 1979: C++ development started by Bjarne Stroustrup.
yes 1970: Xerox PARC founded. 1981: MS-DOS introduced.
no 1970: Xerox PARC founded. 1984: X Windows introduced.
no 1970: Xerox PARC founded. 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee.
no 1970: Xerox PARC founded. 1995: Java programming language and Java VM first released by Sun Microsystems.
no 1972: Smalltalk created. 1965: Butler Lampson's thesis.
yes 1972: Smalltalk created. 1966: Simula introduced by Nygaard and Dahl.
yes 1972: Smalltalk created. 1967: Case Western Reserve University founded.
no 1972: Smalltalk created. 1969: ARPANET started running.
no 1972: Smalltalk created. 1970: Xerox PARC founded.
no 1972: Smalltalk created. 1972: Smalltalk created.
yes 1972: Smalltalk created. 1979: C++ development started by Bjarne Stroustrup.
no 1972: Smalltalk created. 1981: MS-DOS introduced.
yes 1972: Smalltalk created. 1984: X Windows introduced.
yes 1972: Smalltalk created. 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee.
yes 1972: Smalltalk created. 1995: Java programming language and Java VM first released by Sun Microsystems.
no 1979: C++ development started by Bjarne Stroustrup. 1965: Butler Lampson's thesis.
no 1979: C++ development started by Bjarne Stroustrup. 1966: Simula introduced by Nygaard and Dahl.
no 1979: C++ development started by Bjarne Stroustrup. 1967: Case Western Reserve University founded.
yes 1979: C++ development started by Bjarne Stroustrup. 1969: ARPANET started running.
yes 1979: C++ development started by Bjarne Stroustrup. 1970: Xerox PARC founded.
no 1979: C++ development started by Bjarne Stroustrup. 1972: Smalltalk created.
no 1979: C++ development started by Bjarne Stroustrup. 1979: C++ development started by Bjarne Stroustrup.
no 1979: C++ development started by Bjarne Stroustrup. 1981: MS-DOS introduced.
no 1979: C++ development started by Bjarne Stroustrup. 1984: X Windows introduced.
no 1979: C++ development started by Bjarne Stroustrup. 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee.
no 1979: C++ development started by Bjarne Stroustrup. 1995: Java programming language and Java VM first released by Sun Microsystems.
no 1981: MS-DOS introduced. 1965: Butler Lampson's thesis.
no 1981: MS-DOS introduced. 1966: Simula introduced by Nygaard and Dahl.
no 1981: MS-DOS introduced. 1967: Case Western Reserve University founded.
no 1981: MS-DOS introduced. 1969: ARPANET started running.
no 1981: MS-DOS introduced. 1970: Xerox PARC founded.
yes 1981: MS-DOS introduced. 1972: Smalltalk created.
no 1981: MS-DOS introduced. 1979: C++ development started by Bjarne Stroustrup.
no 1981: MS-DOS introduced. 1981: MS-DOS introduced.
no 1981: MS-DOS introduced. 1984: X Windows introduced.
no 1981: MS-DOS introduced. 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee.
no 1981: MS-DOS introduced. 1995: Java programming language and Java VM first released by Sun Microsystems.
no 1984: X Windows introduced. 1965: Butler Lampson's thesis.
yes 1984: X Windows introduced. 1966: Simula introduced by Nygaard and Dahl.
no 1984: X Windows introduced. 1967: Case Western Reserve University founded.
no 1984: X Windows introduced. 1969: ARPANET started running.
no 1984: X Windows introduced. 1970: Xerox PARC founded.
no 1984: X Windows introduced. 1972: Smalltalk created.
no 1984: X Windows introduced. 1979: C++ development started by Bjarne Stroustrup.
no 1984: X Windows introduced. 1981: MS-DOS introduced.
no 1984: X Windows introduced. 1984: X Windows introduced.
no 1984: X Windows introduced. 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee.
yes 1984: X Windows introduced. 1995: Java programming language and Java VM first released by Sun Microsystems.
no 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee. 1965: Butler Lampson's thesis.
no 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee. 1966: Simula introduced by Nygaard and Dahl.
no 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee. 1967: Case Western Reserve University founded.
no 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee. 1969: ARPANET started running.
no 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee. 1970: Xerox PARC founded.
no 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee. 1972: Smalltalk created.
no 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee. 1979: C++ development started by Bjarne Stroustrup.
yes 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee. 1981: MS-DOS introduced.
no 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee. 1984: X Windows introduced.
yes 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee. 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee.
no 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee. 1995: Java programming language and Java VM first released by Sun Microsystems.
no 1995: Java programming language and Java VM first released by Sun Microsystems. 1965: Butler Lampson's thesis.
no 1995: Java programming language and Java VM first released by Sun Microsystems. 1966: Simula introduced by Nygaard and Dahl.
yes 1995: Java programming language and Java VM first released by Sun Microsystems. 1967: Case Western Reserve University founded.
no 1995: Java programming language and Java VM first released by Sun Microsystems. 1969: ARPANET started running.
no 1995: Java programming language and Java VM first released by Sun Microsystems. 1970: Xerox PARC founded.
no 1995: Java programming language and Java VM first released by Sun Microsystems. 1972: Smalltalk created.
no 1995: Java programming language and Java VM first released by Sun Microsystems. 1979: C++ development started by Bjarne Stroustrup.
yes 1995: Java programming language and Java VM first released by Sun Microsystems. 1981: MS-DOS introduced.
no 1995: Java programming language and Java VM first released by Sun Microsystems. 1984: X Windows introduced.
no 1995: Java programming language and Java VM first released by Sun Microsystems. 1989: HTML on the Internet's foundational technology proposed by Tim Berners-Lee.
yes 1995: Java programming language and Java VM first released by Sun Microsystems. 1995: Java programming language and Java VM first released by Sun Microsystems.
{
  hit: {
    '0': 3,
    '1': 1,
    '2': 2,
    '3': 0,
    '4': 1,
    '5': 6,
    '6': 2,
    '7': 1,
    '8': 2,
    '9': 2,
    '10': 3
  },
  notMissed: {
    '0': true,
    '1': true,
    '2': true,
    '3': true,
    '4': true,
    '5': true,
    '6': true,
    '7': true,
    '8': true,
    '9': true,
    '10': true
  }
}
total = 11, hit = 23, missed = 0
```

(this is before I inserted commas between the judgement and two phrases.)